draft-ietf-jsonpath-base.md

```diff
@@ -1400,7 +1400,7 @@ Note that `..` on its own is not a valid selector.
 
 A descendant selector selects zero or more descendants of a node.
 
-A nodelist enumerating the descendants is known as a _descendant nodelist_ when:
+A nodelist enumerating the node and its descendants is known as a _reflexive descendant nodelist_ when:
```

Comment:
Recursive, instead of reflexive?

Reply:
No - I deliberately borrowed the mathematical term "reflexive" because the nodelist includes the node itself.

The definition of _descendant_ is already recursive:

> Descendants (of a node): The children of the node, together with the children of its children, and so forth recursively. More formally, the descendants relation between nodes is the transitive closure of the children relation.

What we have here is the node and its descendants. This is the reflexive closure of the descendants relation. Or, putting it another way, it's the reflexive transitive closure of the children relation. Of course, there's the extra condition about the various possible orderings of the nodelist.

Is there a better phrase?

Reply:
I still don't think "reflexive" is quite right.  It seems like a stretched attempt to come up with a single phrase.

Maybe we just update the "descendants" definition to include the node.  Is there any time we use it to mean strictly children?

Reply:
The binary relation I had in mind is "is a descendant of" which is a relation on JSON values and is the transitive closure of the "is a child of" relation. I considered redefining "descendants" to include the value itself, but that messes up the current descendants selectors because they only select strict descendants of the input value (NB. we don't currently support `..` on its own).

I have pushed a commit which avoids introducing a special term. See if that helps.

Reply:
> Maybe we just update the "descendants" definition to include the node. Is there any time we use it to mean strictly children?

Yes, we decided in the past that "descendants" are children and children of children, not the node itself.

Reply:
> No - I deliberately borrowed the mathematical term "reflexive" because the nodelist includes the node itself.
> 
> The definition of _descendant_ is already recursive:
> 
> > Descendants (of a node): The children of the node, together with the children of its children, and so forth recursively. More formally, the descendants relation between nodes is the transitive closure of the children relation.
> 
> What we have here is the node and its descendants. This is the reflexive closure of the descendants relation. Or, putting it another way, it's the reflexive transitive closure of the children relation. Of course, there's the extra condition about the various possible orderings of the nodelist.
> 
> Is there a better phrase?

XPath 3.1 uses [descendant-or-self](https://www.w3.org/TR/xpath-31/#unabbrev).

JSONAta uses [a few extra words](https://docs.jsonata.org/path-operators#-descendants) to explain

If the draft is targeting an audience of users, "reflexive" may not be the best choice.

Reply:
@danielaparker Please see the updated PR which avoids "reflexive".